server/StudySharp.API/Controllers/CourseController.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System.Threading.Tasks;
+using AutoMapper;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using StudySharp.API.Requests.Courses;
+using StudySharp.API.Responses.Courses;
+using StudySharp.ApplicationServices.Commands;
+using StudySharp.ApplicationServices.Queries;
+using StudySharp.Domain.General;
+
+namespace StudySharp.API.Controllers
+{
+    [Authorize]
+    [ApiController]
+    [Route("api/[controller]")]
+
+    public class CourseController : ControllerBase
+    {
+        private readonly IMediator _mediator;
+        private readonly IMapper _mapper;
+
+        public CourseController(IMediator mediator, IMapper mapper)
+        {
+            _mediator = mediator;
+            _mapper = mapper;
+        }
+
+        [HttpPost("add")]
+        public async Task<OperationResult> Add([FromBody] AddCourseRequest addCourseRequest)
+        {
+            var addCourseCommand = _mapper.Map<AddCourseCommand>(addCourseRequest);
+            return await _mediator.Send(addCourseCommand);
+        }
+
+        [HttpPost("remove")]
+        public async Task<OperationResult> Remove([FromBody] RemoveCourseByIdRequest removeCourseByIdRequest)
+        {
+            var removeCourseByIdCommand = _mapper.Map<RemoveCourseByIdCommand>(removeCourseByIdRequest);
+            return await _mediator.Send(removeCourseByIdCommand);
+        }
+
+        [HttpGet]
+        public async Task<OperationResult> GetCourseById([FromBody] GetCourseByIdRequest getCourseByIdRequest)
```

Comment:
The same as Remove method 
Add [HttpGet("{id:int}")]
Rewrite [FromBody] to [FromRoute]
And add [BindProperty(Name = "id", SupportsGet = true)] to GetCourseByIdRequest model